### ISSUE TITLE
attention tree error

### ISSUE THREAD

**Issue by liuyuanhao** [YRdddream/attattr#7]
Hi YRddream,
I am running your code with your examples and commands. But I found an error when running "attention tree" module. When I use the following command, a blank graph will be generated, and there is an error that the dividend cannot be 0. Could you help me?
Thank u so much~~

!python examples/attribution_tree.py --attr_file "/content/drive/MyDrive/AttentionAttr/attattr/data/model_and_data/model_and_data/mnli_example/attr_zero_base_exp16.json" \
       --tokens_file "/content/drive/MyDrive/AttentionAttr/attattr/data/model_and_data/model_and_data/mnli_example/tokens_and_pred_100.json" \
       --task_name mnli --example_index 16

**Comment by YRdddream** [comment]
Hi,
Can you provide the full trackback of that error?

**Comment by liuyuanhao** [comment]
Thank u so much! Here is the trackback.


[[[ 7.71305893e-04  2.49641265e-05  6.34942920e-06 ...  2.48486731e-05
    1.23885040e-04  1.37973606e-04]
  [-3.58654461e-05  5.79011581e-06  1.87018757e-06 ... -9.48274276e-06
    1.57856713e-05 -3.44107905e-06]
  [ 5.85830396e-05 -3.39297426e-05 -9.12145082e-05 ... -1.00952830e-05
   -1.24251153e-05 -6.63446164e-05]
  ...
  [ 5.51956771e-05 -7.10900564e-05  2.92329960e-05 ... -4.45169043e-04
    1.37960602e-03  6.91399704e-04]
  [ 7.89354528e-05 -8.78587783e-05  4.31111928e-05 ... -8.88694027e-05
    1.00929419e-04  1.94890528e-04]
  [ 6.63459220e-05  2.91172517e-06 -1.66288826e-05 ... -8.92521067e-05
    8.97637081e-05  3.36862479e-04]]

 [[ 3.36997155e-04 -7.95617103e-05 -1.30602675e-05 ... -2.22911587e-05
   -3.87056042e-05  3.86575200e-05]
  [ 7.01076469e-05  5.79100047e-06 -4.46053762e-05 ... -4.78188318e-06
   -5.78423331e-06 -5.38813582e-08]
  [ 2.25710044e-05  1.17796025e-04  2.85711505e-05 ... -1.89121641e-05
    3.01497477e-05 -3.06557010e-05]
  ...
  [-3.65903503e-04  2.13238901e-05  2.00612287e-04 ... -5.61349016e-05
    4.82043292e-04  2.49397199e-04]
  [ 8.73041932e-05  4.76796604e-06  8.42808811e-06 ...  8.29195394e-05
    1.03559644e-05 -3.59116766e-04]
  [ 1.05259431e-05  1.18356234e-06  6.36979563e-06 ...  2.96455420e-06
   -4.46907271e-06 -1.63055366e-06]]

 [[-2.47966500e-06 -2.39916603e-05  1.54110289e-05 ... -3.21951375e-05
   -2.41990870e-05  7.66987477e-05]
  [ 5.29964765e-05  2.85299679e-05  1.82209068e-04 ...  2.37932660e-06
    2.16885437e-05  5.34297917e-05]
  [ 3.59805779e-05  1.27283339e-04 -2.42756996e-04 ...  1.94552557e-05
    8.54474281e-05  5.80936587e-05]
  ...
  [ 1.21029384e-04  1.14816576e-05  1.24787021e-04 ... -1.65286612e-03
    7.76593336e-04 -1.74743286e-04]
  [ 1.96695125e-05 -4.99259840e-06 -1.81981360e-06 ...  1.37044143e-04
   -1.07655031e-04  4.49510362e-05]
  [-5.79370841e-05  5.24399923e-07  9.20064327e-06 ... -1.21499236e-05
   -3.45041138e-05  8.94088372e-06]]

 ...

 [[ 7.69600222e-04  1.37528879e-04  1.25035172e-04 ... -2.88567468e-03
    1.90180324e-04  4.48243634e-04]
  [-9.80619497e-06 -1.16218855e-03 -6.06728221e-05 ...  8.91292421e-05
   -1.24294353e-04  6.20266228e-05]
  [-4.73778378e-06 -2.39957765e-04 -2.90325117e-04 ...  2.39091627e-05
    1.85469367e-04  2.09037180e-04]
  ...
  [-3.27395984e-04  2.55568912e-05  2.85506166e-05 ... -6.23325305e-03
   -2.00375852e-04 -5.11758254e-04]
  [-2.63662826e-06 -1.55023911e-05 -6.59020705e-07 ...  5.59519247e-06
    2.94078314e-05  5.44844781e-05]
  [-3.73519418e-05  3.85403505e-06 -7.48823728e-06 ... -1.71691523e-05
    1.91459964e-04  1.69200978e-04]]

 [[-1.63030331e-04 -1.19757299e-03  1.33156454e-03 ...  1.42491915e-04
    2.30196983e-03  2.66120183e-03]
  [ 9.84641650e-05 -9.94653378e-05  9.62049040e-05 ...  6.85277176e-05
   -6.89164106e-05 -7.54806592e-07]
  [ 2.19570071e-05  8.97392567e-04 -2.36102323e-03 ... -2.14790290e-04
    3.19088887e-04  2.30576610e-04]
  ...
  [ 2.32092755e-04 -4.45670906e-04 -1.90954488e-06 ... -4.15707825e-04
    2.09069694e-05 -2.35642665e-06]
  [-2.92441224e-05 -2.72887859e-05 -5.43988737e-06 ... -3.12417934e-06
    2.56033057e-04  2.96363264e-04]
  [-4.04554851e-05 -2.38609594e-05 -6.03606616e-06 ... -9.91137725e-06
    3.62805570e-04  4.11852159e-04]]

 [[-3.07067988e-03  4.73838503e-04  8.22594830e-03 ...  2.34704374e-03
    2.87645302e-03  3.00982423e-03]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00 ...  0.00000000e+00
    0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00 ...  0.00000000e+00
    0.00000000e+00  0.00000000e+00]
  ...
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00 ...  0.00000000e+00
    0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00 ...  0.00000000e+00
    0.00000000e+00  0.00000000e+00]
  [ 0.00000000e+00  0.00000000e+00  0.00000000e+00 ...  0.00000000e+00
    0.00000000e+00  0.00000000e+00]]]
examples/attribution_tree.py:43: RuntimeWarning: divide by zero encountered in true_divide
  proportion_all[i] /= abs(proportion_all[i][1:,:].max())
examples/attribution_tree.py:43: RuntimeWarning: invalid value encountered in true_divide
  proportion_all[i] /= abs(proportion_all[i][1:,:].max())
examples/attribution_tree.py:47: RuntimeWarning: invalid value encountered in multiply
  proportion_all *= (proportion_all > threshold).astype(int)
<Figure size 3000x2200 with 1 Axes>

And the result.jpg is as follows.
![result](https://user-images.githubusercontent.com/44806441/171132634-df783087-e3ad-4746-8225-278553abb137.jpg)

**Comment by YRdddream** [comment]
Hi, 
I think maybe you didn't use the provided docker.
Please remove the last line of "attr_zero_base_exp16.json" and try again.

**Comment by liuyuanhao** [comment]
Hi YRdddream,
I have deleted the last line but I got the right trackback and a blank result.jpg.
Could u help me?

Here is the trackback.

[[[ 3.58195588e-02  5.23560606e-03  1.68496107e-03 ...  3.04381096e-03
    2.78535712e-03  8.74083159e-04]
  [-2.09696091e-03  1.87435123e-03 -3.87538620e-03 ... -2.26171421e-04
   -6.44006135e-05  1.94411025e-07]
  [-1.88925366e-03 -7.14371250e-03 -1.54210659e-03 ... -8.91131222e-05
    9.34657165e-05 -3.44959961e-05]
  ...
  [ 7.82514114e-03 -8.59185389e-04  8.46923290e-05 ... -2.67159101e-03
    1.53500761e-02 -7.19473752e-04]
  [ 8.13498924e-03  1.59354761e-03  3.51425904e-03 ...  1.05349152e-02
    1.48997673e-03  1.82823683e-02]
  [ 9.38211113e-02  4.38541396e-03  2.37183091e-03 ...  2.10466309e-02
   -9.40656284e-03 -9.78323263e-03]]

 [[ 1.19605622e-01 -1.77603786e-03 -1.45011415e-03 ...  2.24215453e-03
    1.91252432e-03 -9.55576774e-04]
  [-1.17771012e-02 -2.11507803e-03 -2.63051397e-03 ... -1.75240932e-03
    1.08249480e-04 -1.24812619e-03]
  [ 2.18567102e-03 -2.09023365e-03 -1.89336669e-04 ... -8.38968758e-05
   -3.88299621e-05 -1.48360382e-04]
  ...
  [-7.47883171e-02  1.59353286e-03  2.48329974e-04 ... -9.06909882e-03
   -2.97913685e-02 -1.24153280e-01]
  [-1.14118710e-01  3.11476336e-04  2.92613688e-04 ... -3.21106218e-02
   -7.04531305e-03  6.46483305e-02]
  [-1.61130822e-01 -6.37917296e-04 -3.81706945e-05 ...  6.86092757e-02
    1.29797550e-01  2.21905077e-01]]

 [[ 9.60776974e-02 -7.00872160e-03 -3.64160334e-03 ...  1.74299875e-03
    7.21738657e-03 -3.35078464e-03]
  [-7.50852987e-03 -6.34518746e-04  3.04636549e-02 ... -1.34729809e-03
   -3.61234190e-04 -2.87767824e-04]
  [-1.00816219e-02  3.38797805e-04  1.51285729e-04 ... -4.71874774e-04
   -4.68231705e-03  3.96359665e-04]
  ...
  [-3.58453754e-02  1.46608704e-03  1.45910390e-03 ... -4.80631185e-04
   -2.01574156e-02 -2.76656157e-04]
  [-2.34993312e-02 -5.07623408e-03 -4.61906277e-03 ...  1.18050175e-02
   -9.86378935e-04  4.85989412e-02]
  [ 3.30852365e-02 -5.56691991e-03 -4.72051330e-03 ...  3.83752795e-03
    1.84502015e-02 -3.78793620e-03]]

 ...

 [[ 8.27094006e-03  6.92613725e-04 -4.41875772e-05 ... -2.21609459e-04
   -4.99510310e-03  2.85650920e-02]
  [ 6.20296948e-06  1.49723702e-05  1.89760525e-04 ... -1.80678552e-05
    9.70098210e-07  1.12740809e-05]
  [ 4.44094727e-05  1.82229455e-04  1.97976726e-04 ... -4.49642743e-06
    1.53678207e-05 -5.20535656e-06]
  ...
  [ 2.19361725e-03 -6.01052326e-04 -9.56721925e-05 ... -4.28020990e-04
    3.17025057e-04  2.19352307e-04]
  [-9.88352032e-03  2.91122984e-05 -9.69621847e-04 ... -8.49535148e-04
   -1.60410867e-03 -1.96350273e-02]
  [-4.67594942e-03 -9.17549653e-05  1.26237904e-05 ... -6.01255950e-04
   -4.45991876e-03 -5.90915535e-02]]

 [[ 1.34859525e-01 -7.40859326e-04 -6.84708423e-04 ...  3.23272863e-03
   -2.48270016e-02 -1.21426785e-01]
  [ 3.35565362e-04 -1.67663903e-04  2.56928844e-04 ... -2.36926493e-05
   -1.13593249e-04 -5.55357861e-05]
  [ 4.86054817e-04 -1.35014343e-04 -5.76383677e-04 ... -1.52198929e-05
   -3.40803651e-04 -2.49436192e-04]
  ...
  [-8.98298631e-04 -1.15078477e-04 -1.18738396e-04 ...  2.21277947e-03
    3.43527795e-04 -4.55989350e-04]
  [-1.36089021e-02  1.80743034e-04  6.93461877e-05 ...  5.29747808e-03
    5.89764178e-03  5.52084875e-03]
  [-3.78634174e-03  1.03847898e-03  6.01582426e-04 ...  4.48869129e-03
   -4.22223442e-03 -8.19741300e-03]]

 [[ 1.78202503e-01 -3.53736029e-04 -1.51545081e-03 ...  9.96079905e-04
    2.76602097e-02  5.98121656e-02]
  [-9.44514688e-05  1.15613056e-04 -3.14205630e-05 ... -3.54526296e-06
   -5.69716056e-05  5.96455326e-06]
  [-9.04244367e-04 -8.24075069e-05 -5.88428491e-04 ... -6.80341314e-05
   -2.48806630e-04  7.53717997e-05]
  ...
  [-9.02535670e-04  7.84791642e-05  8.27927055e-05 ... -1.65515446e-04
   -1.32204938e-03 -1.02078227e-03]
  [ 4.53831306e-03  1.56064839e-04  1.36567764e-04 ...  2.46492729e-04
   -9.68444910e-03 -1.32352465e-02]
  [ 4.24265391e-03  9.92532831e-04  1.15086766e-03 ...  6.14262376e-05
   -1.58295743e-02 -3.74012999e-02]]]
<Figure size 3000x2200 with 1 Axes>

And the following is the result.jpg:
![image](https://user-images.githubusercontent.com/44806441/171176013-237c9bdf-2ebe-4588-9757-4445115d4e62.png)

Sorry, I still can not find the wrong issue, could u help me? Thank u for your previous help.
